bundles/org.platformio.eclipse.ide.installer/src/org/platformio/eclipse/ide/installer/Installer.java

```diff
@@ -14,38 +14,39 @@
 
 import java.io.IOException;
 import java.util.Optional;
+import java.util.stream.Stream;
 
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IConfigurationElement;
 import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Platform;
 import org.platformio.eclipse.ide.installer.api.Environment;
 import org.platformio.eclipse.ide.installer.api.PythonsRegistry;
 import org.platformio.eclipse.ide.installer.base.BaseEnvironment;
 import org.platformio.eclipse.ide.installer.piocore.LocalPioCoreDistribution;
 import org.platformio.eclipse.ide.installer.piocore.PioCoreDistribution;
+import org.platformio.eclipse.ide.installer.python.LocalPython;
 import org.platformio.eclipse.ide.installer.python.Python;
 import org.platformio.eclipse.ide.installer.python.PythonDistribution;
 
 public final class Installer {
 
 	private final Environment environment = new BaseEnvironment();
-	private final PythonsRegistry registry;
+	private Python python;

```

Comment:
mutable fields should be avoided
you can have a final List here
and then go throw this list in `killPio` method